### PR TITLE
Stavan/ch3

### PR DESCRIPTION
**Pull request by Stavan-Jain**
The main thing to look at is the second commit. I added a lot of comments to the beginning of sheet 3. I'm not entirely satisfied with the explanation. It has lots of room for improvement, but I really do think it's something the users should know/explore.

### CODE REVIEW

File: AlgebraInLean/Chapter03/Sheet02.lean [hunk 1]
```diff
@@ -4,12 +4,13 @@ namespace Defs
...
-group operation is multiplication, the functions we define in this sheet are equivalent to
-exponentiation.
+## Challenge Sheet
 
-First, we define the power function `mpow` for monoids. Since monoids do not have a notion of
-inverses, we consider only natural numbers as input.
+In this sheet, we build the theory of repeated application of the group operation. If the group
+operation is multiplication, the functions we define in this sheet are equivalent to exponentiation.
+
+First, we define the power function `mpow` for monoids. Let M be a Monoid and m ∈ M. Since monoids
```
Comment (MAO3J1m0Op) [inline AlgebraInLean/Chapter03/Sheet02.lean]: Should be x \in M

File: AlgebraInLean/Chapter03/Sheet03.lean [hunk 2]
```diff
@@ -4,12 +4,55 @@ namespace Defs
...
+above is either.
+
+You may object to this: if `isFiniteOrder x` is true, then the expression is the natural number
+returned by Nat.find and if `isFiniteOrder x` is false, then the expression is the natural number 0.
+In either case, the expression has type `ℕ`. Shouldn't Lean be able to figure this out?
+
+Implicit in this intuition is the idea that every proposition is either true or false, even if we
+don't necessarily have a way to compute the result. In other words, for any proposition `p`, either
+`p` is true or its negation `¬p` is true. However, this seemingly trivial claim, called the Law of
+the Excluded Middle, is not true in constructive mathematics! Moreover, Lean's foundations for
```
Comment (MAO3J1m0Op) [inline AlgebraInLean/Chapter03/Sheet03.lean]: I'd say "not always true"

File: AlgebraInLean/Chapter03/Sheet03.lean [hunk 3]
```diff
@@ -4,12 +4,55 @@ namespace Defs
...
+proving theorems is constructive: if you want to prove a proposition `p`, you have to construct a
+term that has type p (i.e., a term of type `p` is a proof of `p`). What we need is classical logic!
```
Comment (MAO3J1m0Op) [inline AlgebraInLean/Chapter03/Sheet03.lean]: To do the transition, I'd get rid of the "What we need is classical logic!", add a "Meanwhile, " before the paragraph below, and maybe stitch these two paragraphs together.

**Review by MAO3J1m0Op — APPROVED** [review]
We'll keep tweaking minor wordings, but these changes are very welcome!

**Comment by Stavan-Jain** [comment]
Sorry for the crazy commit history. I rebased on #18 and force pushed so that I could look at and potentially edit the completed gpow section :)

**Comment by Stavan-Jain** [comment]
You can merge if you'd like. Lots more to do after the presentation lol

**Pull request merged**